src/pages/DailyStatus.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect, useCallback } from "react";
+import "../App.css";
+import "./home.css";
+import { DAILY_STATUS } from "../constants/constants";
+
+const DailyStatus = ({
+  state,
+  setState,
+  setPage,
+  page,
+}) => {
+  const [isSame, setIsSame] = useState(false);
+  const [isBetter, setIsBetter] = useState(false);
+  const [isWorse, setIsWorse] = useState(false);
+  const [statusError, setStatusError] = useState(false);
+
+  const handleOnStatusClick = useCallback(
+    (item) => () => {
+      const newState = { ...state, statusToday: item };
+      const isStatusChecked = Object.values(newState).some((s) => s);
```

Comment:
please move `Object.values(newState).some((s) => s);` to a separate function and reuse in onNext:
`const getIsStatusChecked = (state) => Object.values(state).some((s) => s);`

`const onNext = useMemo(async () => {
    const isStatusChecked = getIsStatusChecked(state);
    setStatusError(!isStatusChecked);
    if (isStatusChecked) {
        setPage(page + 1);
    }
  }, [state, page]);`

Reply:
resolved